src/components/Grid/Grid.css

```diff
@@ -0,0 +1,270 @@
+/* =======================================================================
+  Responsive 12 column grid using css grid.
+
+  column = 60px;
+  gutters = 30px;
+  rows = auto;
+
+  By default the grid is 960px wide and centered with margin: 0 auto;
+  as it scales down the column width get's smaller. 
+========================================================================== */
+
+.grid-container {
+  max-width: 960px;
+  padding: 0 1.25em;
+  display: grid;
+  grid-auto-flow: row;
+  align-items: center;
+  flex-wrap: wrap;
+
+  /* 12 equal colums each taking up the available space 1fr (fraction unit) */
+  grid-template-columns: repeat(12, 1fr);
+
+  /* 20px grid gap */
+  gap: 1.25rem;
+
+  /* collapse all gutters */
+  &--collapsed {
+    grid-gap: 0rem;
+    padding: 0;
+
+    /* collapse horizontal guttes only */
+    &-cells {
+      padding: 0;
+      grid-row-gap: 1.25rem;
+      grid-column-gap: 0rem;
+    }
+
+    /* collapse vertical guttes only */
+    &-rows {
+      grid-row-gap: 0rem;
+      grid-column-gap: 1.25rem;
+    }
+  }
+
+  /* make full width of container */
+  &--fullWidth {
+    max-width: 100%;
+  }
+
+  /* nested grid containers */
+  & .grid-container {
+    @inherit .mx-0, .px-0;
+  }
+}
+
+/* 
+ * Rows .row-N - unlimited number
+ * 
+ * Rows have an implicit height but you can use the row class to 
+ * change the presentation order of cells 
+   ========================================================================== */
+@variants responsive {
```

Comment:
What are the use cases for this? Is it worth adding 100 classes to the css?

Reply:
use cases for responsive rows are that you can change source order for mobile, think about needing to shift a sidebar to the bottom of the page on mobile

Reply:
we can always run the css through https://cssnano.co/ in our consuming apps to trim the fat